data/src/main/java/org/apache/iceberg/data/InternalRecordWrapper.java

```diff
@@ -80,10 +80,15 @@ public int size() {
 
   @Override
   public <T> T get(int pos, Class<T> javaClass) {
-    if (transforms[pos] != null) {
-      return javaClass.cast(transforms[pos].apply(wrapped.get(pos, Object.class)));
+    Object value = wrapped.get(pos, Object.class);
+
+    if (value == null) {
+      return null;
```

Comment:
Why is it incorrect to call `javaClass.cast(null)`?

Reply:
If we don't make this change,  when running this unit tests we will encounter the exception: 

```java
temporal
java.lang.NullPointerException: temporal
	at java.util.Objects.requireNonNull(Objects.java:228)
	at java.time.LocalDate.from(LocalDate.java:364)
	at java.time.LocalDate.until(LocalDate.java:1602)
	at java.time.temporal.ChronoUnit.between(ChronoUnit.java:272)
	at org.apache.iceberg.util.DateTimeUtil.daysFromDate(DateTimeUtil.java:42)
	at org.apache.iceberg.data.InternalRecordWrapper.lambda$converter$2(InternalRecordWrapper.java:48)
	at org.apache.iceberg.data.InternalRecordWrapper.get(InternalRecordWrapper.java:84)
	at org.apache.iceberg.types.Comparators$StructLikeComparator.compare(Comparators.java:122)
	at org.apache.iceberg.types.Comparators$StructLikeComparator.compare(Comparators.java:102)
	at org.apache.iceberg.util.StructLikeWrapper.equals(StructLikeWrapper.java:76)
	at org.junit.Assert.isEquals(Assert.java:131)
	at org.junit.Assert.equalsRegardingNull(Assert.java:127)
	at org.junit.Assert.assertEquals(Assert.java:111)
	at org.apache.iceberg.spark.source.TestInternalRowWrapper.generateAndValidate(TestInternalRowWrapper.java:71)
	at org.apache.iceberg.RecordWrapperTest.generateAndValidate(RecordWrapperTest.java:112)
	at org.apache.iceberg.RecordWrapperTest.testSimpleStructWithoutTime(RecordWrapperTest.java:62)
	at java.lang.Thread.run(Thread.java:748)
```

As you said ,  `javaClass.cast(null)` is OK , but the key problem is the `transforms[pos].apply(null)`.  we have many `transforms` which don't allow to handle `null` values.